app-typescript/components/Dropdown.tsx

```diff
@@ -0,0 +1,270 @@
+        return function() {
+            const context = dropdownElements;
+            clearTimeout(inDebounce);
+            inDebounce = setTimeout(() => calculate.apply(context), delay);
+        };
+    };
+
+    function elementAlign() {
+        if (align) {
+            return align === 'right' ? 'right' : '';
+        } else {
+            return width ? 'right' : '';
+        }
+    }
+
+    function each(item: any, index: number) {
+        if (item['type'] === 'submenu') {
+            let submenuItems: any = [];
+            item['items'].forEach((el: any, key: number) => {
+                submenuItems.push(each(el, key));
+            });
+            return (
+                <li key={index}>
+                    <div className={(submenu ? 'dropdown--dropup' : '') + ' dropdown'} >
+                        <button
+                            className='dropdown__toggle dropdown-toggle'
+                            onMouseOver={() => submenuItems.map((element: any) => { calculateTwo.apply(element); })} >
+                            {item['icon'] ? <i className={'icon-' + item['icon']}></i> : null}
+                            {item['label']}
+                        </button>
+                        <ul className={(width ? 'dropdown__menu--submenu-left ' : '') + 'dropdown__menu'} ref={ref}>
+                            {submenuItems}
+                        </ul>
+                    </div>
+                </li>
+            );
+
+        } else if (item['type'] === 'group') {
+            let groupItems: any = [];
+            item['items'].forEach((el: any, key: number) => {
+                groupItems.push(each(el, key));
+            });
+            return (
+                <React.Fragment key={index}>
+                    <li>
+                        <div className="dropdown__menu-label">{item['label']}</div>
+                    </li>
+                    {groupItems}
+                </React.Fragment>
+            );
+
+        } else if (item === 'divider') {
+            return (<li className="dropdown__menu-divider" key={index}></li>);
+        } else {
+            return (
+                <DropdownItem
+                    key={index}
+                    label={item['label']}
+                    icon={item['icon']}
+                    onSelect={item['onSelect']} />);
+        }
+    }
+
+    const headerElements = header?.map((el, index) => {
+        return each(el, index);
+    });
+
+    const dropdownElements = items.map((el, index) => {
+        return each(el, index);
+    });
+
+    const footerElements = footer?.map((el, index) => {
+        return each(el, index);
+    });
+
+    const classes = classNames('dropdown', {
+        ['open']: open,
+        ['dropdown--dropup']: heightSet(height),
+        ['dropdown--align-right']: elementAlign() === 'right',
+    });
+
+    return (
+        <div className={classes} >
+            {typeof children === 'object' ?
```

Comment:
<Dropdown ...>
    <i className="some btn"></i>
</Dropdown> 
In this case it is true, and in this case
<Dropdown ...>
    Button name
</Dropdown> 
it's not. In case you want default style of button you just write 'string' as a child of Dropdown component and if you want a styled button for example like navigation button then as a child of component send that navigation button as the element. 